server/src/main/java/chat/engine/Connect.java

```diff
@@ -108,13 +114,27 @@ private void closeConnection() throws IOException {
         reader.close();
         writer.close();
         clientsOnline.remove(name);
+    }
+
+    private void sendMessage(String message, String name) throws IOException {
 
+        for (Map.Entry<String, Socket> entry : clientsOnline.entrySet()) {
+            if (entry.getValue() != socket) {
+                writer = getWriter(entry.getValue());
+                writer.println(name + ": " + message);
+            }
+        }
     }
 
-    private void sendMessage(String message) throws IOException {
+    private boolean privateMessage(String message, String name) throws IOException {
+        String namePM = "";
         for (Map.Entry<String, Socket> entry : clientsOnline.entrySet()) {
```

Comment:
Do you understand what do you do here?
You iterate through map with online users, and for each user in this map checking is your message contains valid structure, for what?